superset/assets/javascripts/explore/components/controls/SelectAsyncControl.jsx

```diff
@@ -0,0 +1,62 @@
+/* global notify */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Select from '../../../components/AsyncSelect';
+import { t } from '../../../locales';
+
+const propTypes = {
+  onChange: PropTypes.func,
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+    PropTypes.arrayOf(PropTypes.string),
+    PropTypes.arrayOf(PropTypes.number),
+  ]),
+};
+
+const defaultProps = {
+  onChange: () => {},
+};
+
+class SelectAsyncControl extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Many of the other controls to not maintain internal state and rely solely on the `value` prop and `onChange` to act as a controlled component. There's nothing bad with maintaining the state here but it adds a bid of complexity that is not needed.

Reply:
fixed.